### PR TITLE
WIP: Add AWS RDS docs

### PR DESCRIPTION
**Pull request by jacobbednarz**
This adds a new file under the `doc` directory that covers the caveats
and work arounds for using Amazon RDS.

The idea here is that we have a RDS document and each engine (mysql,
aurora, etc) gets a sub section for specific stuff and everything else
ends up as general as possible.

### CODE REVIEW

**Comment by jacobbednarz** [comment]
@pbitty @shlomi-noach do you mind running your eyes over this and making sure I've covered everything?

File: doc/rds.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Amazon RDS
```
Comment (shlomi-noach) [inline doc/rds.md]: Let's just add a quick sentence before this line, carifying that gh-ost actually runs on RDS and Aurora.

File: doc/rds.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Limitations
+
+- No `SUPER` privileges.
+- `gh-ost` runs should be setup use [`--assume-rbr`][assume_rbr_docs] and use `binlog_format=ROW`.
```
Comment (shlomi-noach) [inline doc/rds.md]: `[assume_rbr_docs]` link

File: doc/rds.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+- Aurora does not allow editing of the `read_only` parameter. While it is defined as `{TrueIfReplica}`, the parameter is non-modifiable field.
+- In order to have binlogs enabled, the backup window must be set to greater than 1 day.
+
+## Aurora
+
+#### Replication
+
+In Aurora replication, you have separate reader and writer endpoints however because the cluster shares the underlying storage layer, `gh-ost` will detect it is running on the master. This becomes an issue when you wish to use [migrate/test on replica][migrate_test_on_replica_docs] because you won't be able to use a single cluster in the same way you would with MySQL RDS.
+
+To work around this, you can follow along the [AWS replication between clusters documentation][aws_replication_docs] for Aurora with one small caveat. For the "Create a Snapshot of Your Replication Master" step, the binlog position is not available in the AWS console. You will need to issue the SQL query `SHOW SLAVE STATUS` or `aws rds describe-events` API call to get the correct position.
```
Comment (shlomi-noach) [inline doc/rds.md]: `[aws_replication_docs]` link

File: doc/rds.md (hunk 2 above)
Comment (jacobbednarz) [inline doc/rds.md]: I gather you mean, "whoa, where's the actual link? have you gone crazy?" 😛 

These are [reference links](https://daringfireball.net/projects/markdown/syntax#link) which allow you to reuse a single link multiple times without repeating it over and over. The links themselves are put at the [bottom](https://github.com/github/gh-ost/pull/385/files#diff-82208f0a804594829ab45d4f59e7d19fR28).

Or would you rather not use reference links?

Comment (shlomi-noach) [inline doc/rds.md]: oh, you're right 😛 , thank you for pointing this out!

**Comment by jacobbednarz** [comment]
> However, can we have a playbook checklist available for Aurora? A list of actions/configurations that must be checked/set for running the migration?

Sure! Let me ponder on how to best present this without confusing people who actually want to use non-RDS documentation when they come searching for it.

**Comment by shlomi-noach** [comment]
@jacobbednarz I changed the title to "WIP: ..." (Work In Progress). Please ping me when you wish me to merge this PR. Thank you very much!

File: doc/rds.md (hunk 1 above)
Comment (jacobbednarz) [inline doc/rds.md]: Fixed in 15d30ce

**Comment by jacobbednarz** [comment]
Thanks @shlomi-noach. I'm standing up fresh clusters at the moment to put together a checklist and should be able to report back in the next day or so with some updates.

**Comment by jacobbednarz** [comment]
@pbitty do you mind casting your 👀  over 55d8b3a and letting me know if you think anything is missing? Much of the stuff I have in my list is internal to us and I'm unsure how much knowledge assumption we should be making here.

**Comment by pbitty** [comment]
Sorry for the late reply. I will take a look at it today.

**Comment by jacobbednarz** [comment]
@shlomi-noach I'm happy to merge as is and if there is any feedback from @pbitty (or others on RDS) we can open new PRs for it.

**Comment by jacobbednarz** [comment]
Oh, derpp! :man_facepalming: Let me add that too.

**Comment by jacobbednarz** [comment]
Are you 👌 with c3eda3f?

**Comment by shlomi-noach** [comment]
Thank you!

**Comment by jacobbednarz** [comment]
🎉  🎉  🎉

**Pull request merged**